Date & time util documentation (#397)
Refinitiv/refinitiv-ui: documents/src/pages/utils/date-time.md

layout: default
-->

::status-complete::

# Date & Time

> The utility provides the comprehensive set of tools to manipulate with date/time/datetime strings. It overcomes the problems with JavaScript `Date` implementation, such as:

Reviewer: ```suggestion
The utility provides a comprehensive set of tools to manipulate _date_, _time_ and _datetime_ strings. It overcomes the problems with JavaScript's `Date` implementation, such as:
```

Author: Fixed

> - string to date parsing is error-prone and can be different in browsers

Reviewer: ```suggestion
- String to date parsing is error-prone and can be different in browsers
```

Author: Fixed

- Date object is mutable

> - daylight saving behavior is unpredictable

Reviewer: ```suggestion
- Daylight saving behavior is unpredictable
```

Author: Fixed

> - computation APIs are unwieldy or unavailable

Reviewer: ```suggestion
- Computation APIs are unwieldy or unavailable
```

Author: Fixed

> - defining and parsing dates in _UTC_, _Local_ or _Developer Defined_ timezones is error-prone

Reviewer: ```suggestion
- Defining and parsing dates in _UTC_, _Local_ or _Developer Defined_ timezones is error-prone
```

Reviewer: I'm seeing a lot of your descriptions not starting capitalised. Is this on purpose? If so, how come?

Author: previous sentence ends with semi column (`...implementation, such as:`), it means that the list continues and should not be capitalised. `Date` is capitalised because it is the name of the object

Reviewer: Ok what about descriptions in tables?

Author: Same, if the name, than capitalised, e.g. `Number` or `String`, otherwise lower case

Reviewer: It looks odd for some reason. Just looked at what Lit are doing and they Capitalise each line.

Reviewer: ![image](https://user-images.githubusercontent.com/1519516/179698886-10fe47dc-9579-4710-b873-b087198055e6.png)

Author: date-fns does not

Author: I think it is the matter of consistency really, and for consistency with other pages I will capitalise them :)

Author: Fixed

The utility comes with the ability to parse localised dates and has over 40 functions for date manipulation.

```typescript
import { Locale, addUnit } from '@refinitiv-ui/utils/date.js';

// Create locale object for en-GB
const locale = Locale.fromOptions({ dateStyle: 'short', timeStyle: 'short' }, 'en-GB');

// Parse localised date
const date = locale.parse('21/09/2022, 09:21'); // => '2022-09-21T09:21'

// Add 5 months
addUnit(date, 'month', 5); // => '2023-02-21T09:21'
```

## Format Tokens

> The utility works with the following selection of `ISO8601` date string formats (called as format tokens).

Reviewer: ```suggestion
The utility works with the following selection of `ISO8601` date string formats (format tokens).
```

Author: Fixed
